analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -122,19 +126,33 @@ internal fun SpdxExternalDocumentReference.getSpdxPackage(packageId: String, wor
 /**
  * Return the [SpdxDocument] this [SpdxExternalDocumentReference]'s [SpdxDocument] refers to.
  */
-private fun SpdxExternalDocumentReference.resolve(workingDir: File): SpdxDocument {
+private fun SpdxExternalDocumentReference.resolve(workingDir: File, issues: MutableList<OrtIssue>): SpdxDocument {
     val uri = runCatching { URI(spdxDocument) }.getOrElse {
         throw IllegalArgumentException("'$spdxDocument' identified by $externalDocumentId is not a valid URI. }")
     }
 
-    if (uri.scheme.equals("file", ignoreCase = true) || !uri.isAbsolute) {
-        val referencedFile = workingDir.resolve(uri.path)
-        val spdxFile = referencedFile.takeIf { it.isFile }
+    var tempDir: File? = null
+
+    val spdxFile = if (uri.scheme.equals("file", ignoreCase = true) || !uri.isAbsolute) {
+        val referencedFile = workingDir.resolve(uri.path).normalize()
+        referencedFile.takeIf { it.isFile }
             ?: throw IllegalArgumentException("The local file URI '$uri' does not point to an existing file.")
-        return SpdxModelMapper.read(spdxFile)
+    } else {
+        tempDir = createOrtTempDir()
+        OkHttpClientHelper.downloadFile(uri.toString(), tempDir).getOrThrow()
+    }
+
+    val hash = Hash.create(checksum.checksumValue, checksum.algorithm.name)
+    val verified = hash.verify(spdxFile)
+    if (!verified) {
+        val referencedFile = tempDir?.let { uri } ?: spdxFile
+        issues += createAndLogIssue(
+            source = externalDocumentId,
+            message = "The SPDX file '$referencedFile' does not match the expected $hash."
+        )
```

Comment:
Please make the message unique, e.g. such that it can be matched unambiguously.
IMO it would make sense to include the definition file path relative to the analyzer root in the error message.

Reply:
The definition file path is not known at this point, and making it known would require quite some refactoring. Any other proposals what property to use?

Reply:
Don't have any proposal right now.
I'd be ok with a TODO comment as well, if that works for you.

Reply:
I've reworked the code to be able to refer to the `definitionFile` in the message.

Reply:
Thanks!